### PR TITLE
EXTCODEHASH Opcode Implementation

### LINKED ISSUE #356: Implement EXTCODEHASH opcode
Related EIP: https://github.com/ethereum/EIPs/blob/master/EIPS/eip-1052.md

### PR DESCRIPTION
**Pull request by mersinvald**
Related EIP: https://github.com/ethereum/EIPs/blob/master/EIPS/eip-1052.md
Closes #356

### CODE REVIEW

**Comment by mersinvald** [comment]
TODO: cross-check with geth implementation

File: src/commit/account.rs [hunk 1]
```diff
@@ -438,17 +438,26 @@ impl<A: AccountPatch> AccountState<A> {
     /// Find code by its address in this account state. If the search
     /// failed, returns a `RequireError`.
     pub fn code(&self, address: Address) -> Result<Rc<Vec<u8>>, RequireError> {
+        let code = self.code_opt_nonexist(address)?
```
Comment (r8d8) [inline src/commit/account.rs]: "?.unwrap_or_else()" - not a best practice, 
better to use "?" just on the end of expression, and use explicit methods inside chained calls.

Comment (mersinvald) [reply src/commit/account.rs]: Agreed. Fixed that.

**Review by r8d8 — APPROVED** [review]
LGTM

**Review by tzdybal — APPROVED** [review]
LGTM

**Pull request merged**